src/shared/sam/activation.ts

```diff
@@ -132,6 +135,24 @@ async function activateCodeLensProviders(
         telemetryService
     }
 
+    disposables.push(
+        vscode.languages.registerCodeLensProvider(
```

Comment:
One of the nice things about the recent debugger changes is that they have been done in piecemeal fashion instead of working in a feature branch, which would result in a larger merge when it is complete. So far most of the changes are dormant. This is the first change that users would notice (new codelenses) when the toolkit is next released. Clicking the codelens will create a launch configuration (great!) however running the launch configuration produces a "not implemented" message, which would be confusing to users.

Would it be reasonable to gate this behind the feature flag system (as releasable code) until we have some implementations for the aws-sam debug configurations?

Reply:
FYI re: feature toggles: https://github.com/aws/aws-toolkit-vscode/blob/master/designs/feature-toggle.md

Reply:
I'm wondering if feature toggles are better suited for features that actually "work" but may have bugs.

Considering we know this one doesn't work, perhaps we should keep a side branch going. When disabled, it would have really no impact on the user so we might not have much to gain by putting it out into the wild versus keeping multiple incremental changes in a side branch and merging later.

Reply:
We've had previous feature branches that fell out of sync, and the resulting merge back to mainline didn't go so well. The nice thing about the feature flag is when everything is ready to go, enabling it becomes a non-event (removing the feature flag)

Reply:
:+1: for feature toggles if this is merged directly into `master`.
:-1: for feature toggles if this is merged into `feature/debugconfig` instead  (which is likely) since we're already pruning existing codelens functionality from that branch.